include/snowflake/platform.h

```diff
@@ -56,21 +56,13 @@ void STDCALL sf_tzset(void);
 
 int STDCALL sf_setenv(const char *name, const char *value);
 
-/* on Windows, this function allocate new memory, caller should free it */
-char *STDCALL sf_getenv(const char *name);
+char *STDCALL sf_getenv(const char *name, char *outbuf, size_t bufsize);
```

Comment:
we are changing the public API of libsnowflakeclient - should we do this?

Reply:
This is utility function used in libsnowflakeclient internally.
I don't think external application would rely on that (like sb_ functions, it's available in standard c lib getenv, strcpy, strcat...)
Here we don't have perfect secure replacement for getenv cross all platforms with exactly the same interface.
Having buffer allocation internally would require free on application side and have thread-safe issue as well.
I think changing the interface to force the application take care of the buffer is a better option compare to commenting in header file to ask for free after calling this function.

Reply:
keep the existing interface unchanged and mark it deprecated to keep the backward compatibility.

Reply:
there are no depractions in this file - the old methods are still missing

Reply:
platform.h includes Simba_CRTFunctionSafe.h at the top so it can get deprecations from there.
https://github.com/snowflakedb/libsnowflakeclient/pull/659/files#diff-25951124eb05ec39b13d5c29ce2c708d01d9363367ee8e7c1a946e1774cddc6eL49